packages/axelar-core-std/src/nexus/execute.rs

```diff
@@ -40,8 +40,9 @@ fn parse_message_id(message_id: &str) -> Result<(nonempty::Vec<u8>, u64), Error>
 impl From<router_api::Message> for Message {
     fn from(msg: router_api::Message) -> Self {
         // fallback to using all 0's as the tx ID if it's not in the expected format
+        let zeroed_tx_id = vec![0; 32].try_into().expect("conversion should not fail");
```

Comment:
```suggestion
        let zeroed_message_id = vec![0; 32].try_into().expect("conversion should not fail");
```
message_id != tx_id (at least not all the time)